### PR TITLE
Auto sidebar

### PR DESCRIPTION
**Pull request by mattheu**
Instead of using a nav menu, we should generate this automatically.

The big change is that we now rely on the menu_order to set the order of the pages, so will need to be updated once this goes live.

I wonder if anyone has used any good plugins that make the editing experience on the 'pages' admin screen a bit better for dealing with hierarchical pages?

### CODE REVIEW

**Comment by mattheu** [comment]
Merged private post handling into this, as we need to do it slightly differently now.

**Comment by willmot** [comment]
I've used https://wordpress.org/plugins/simple-page-ordering/ a few times, is pretty good

**Comment by mattheu** [comment]
Updated to revert removal of setting a nav menu, the full tree navigation is now just a default if no nav menu is assigned to that location.

**Comment by mattheu** [comment]
Plugin is already on HM network. I enabled + ordered the pages. I'm going to merge this one.

File: inc/primary-nav.php [hunk 1]
```diff
@@ -4,6 +4,69 @@
...
+
+}
+
+/**
+ * Output a single page tree navigation list item.
+ *
+ * @param  WP_Post $page
+ * @return null
+ */
+function render_nav_item( WP_Post $page ) {
```
Comment (willmot) [inline inc/primary-nav.php]: This is causing a Fatal on `catalog.hmn.md`

```
Error: Argument 1 passed to HM_Handbook\render_nav_item() must be an instance of HM_Handbook\WP_Post, instance of WP_Post given
```

Shouldn't it be 

```PHP
function render_nav_item( \WP_Post $page ) {
```

**Pull request merged**